src/databricks/labs/ucx/hive_metastore/catalog_schema.py

```diff
@@ -144,8 +144,10 @@ def _validate_location(self, location: str):
         for external_location in self._external_locations:
             if location == external_location.url:
                 return True
-            if location_path.match(f"{external_location.url}/*"):
-                return True
+            if external_location.url is not None:
+                external_location_path = PurePath(external_location.url)
+                if location_path.parts[: len(external_location_path.parts)] == external_location_path.parts:
+                    return True
```

Comment:
Does this work?
```suggestion
            if location_path.match(f"{external_location.url}/**"):
                return True
```

Reply:
@JCZuurmond This doesn't seem to work. However, this works `location_path.match(f"{external_location}/*/*")`. Again, this fails if another additional level of nesting is provided.

Reply:
hmmm, too bad, could you add a test for a deeply nested structure does not work? Like 8 levels deep

Reply:
`location_path.match(f"{external_location}/**/*")` What does that do? I would expect it to be solvable with glob pattern

Reply:
@JCZuurmond Looks like ** is not supported by PurePath.match. However, I am able to get a match using fnmatch. `fnmatch.fnmatch(location, external_location*)`. Please review.